### PR TITLE
Argparse

### PR DESCRIPTION
**Pull request by syclops**
I changed several files that examine the number of command-line arguments in `sys.argv` and replaced them with use of the `argparse` module. This is a cleaner, more extensible way to process command-line arguments without having to do all the error handling ourselves. (The only things it's buying us right now is the ability to log the error message.)
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/275%23issuecomment-127215473%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23issuecomment-127232684%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078652%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078868%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078905%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078943%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078968%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36079163%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36083919%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36083920%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36083921%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36083922%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23issuecomment-127215473%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22A%20few%20minor%20comments%20aside%2C%20very%20happy%20to%20see%20this%20change%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-08-03T12%3A15%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-08-03T13%3A24%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20512f8ea9f15acbcc59ef180ff506be01b725943a%20infrastructure/beacon_server.py%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078868%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20keep%20imports%20sorted%20%28with%20the%20%27import%20x%27%20form%20coming%20before%20the%20%27from%20x%20import%20y%27%20form%29.%20Same%20applies%20to%20other%20files.%22%2C%20%22created_at%22%3A%20%222015-08-03T12%3A11%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-03T13%3A22%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/beacon_server.py%3AL24-31%22%7D%2C%20%22Pull%20512f8ea9f15acbcc59ef180ff506be01b725943a%20infrastructure/dns_server.py%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078943%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Should%20be%20%60%27server_id%27%60%20for%20consistency%22%2C%20%22created_at%22%3A%20%222015-08-03T12%3A12%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-03T13%3A22%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/dns_server.py%3AL504-517%22%7D%2C%20%22Pull%20512f8ea9f15acbcc59ef180ff506be01b725943a%20infrastructure/dns_server.py%2017%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078968%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%60%27DNS%20Domain%27%60%20would%20be%20slightly%20clearer%22%2C%20%22created_at%22%3A%20%222015-08-03T12%3A12%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-03T13%3A22%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/dns_server.py%3AL504-517%22%7D%2C%20%22Pull%20512f8ea9f15acbcc59ef180ff506be01b725943a%20infrastructure/beacon_server.py%2040%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/275%23discussion_r36078652%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20isn%27t%20necessary%20any%20more%3A%5Cr%5Cn%60%60%60%5Cr%5Cnbeacon_server.py%3A%20error%3A%20argument%20type%3A%20invalid%20choice%3A%20%27locale%27%20%28choose%20from%20%27core%27%2C%20%27local%27%29%5Cr%5Cn%60%60%60%22%2C%20%22created_at%22%3A%20%222015-08-03T12%3A07%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Ah%2C%20good%20point.%20Should%20I%20then%20use%20a%20simple%20if-else%20%28and%20just%20check%20for%20%60core%60%29%20or%20have%20an%20if-elif%20as%20above%3F%22%2C%20%22created_at%22%3A%20%222015-08-03T12%3A11%3A42Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1296929%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/syclops%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20think%20a%20simple%20if-else%20should%20be%20fine%22%2C%20%22created_at%22%3A%20%222015-08-03T12%3A15%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-08-03T13%3A22%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20infrastructure/beacon_server.py%3AL1359-1381%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/kormat%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/kormat'></a>
- [x] <a href='#crh-comment-Pull 512f8ea9f15acbcc59ef180ff506be01b725943a infrastructure/beacon_server.py 40'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/275#discussion_r36078652'>File: infrastructure/beacon_server.py:L1359-1381</a></b>
- <a href='https://github.com/kormat'></a> This isn't necessary any more:

```
beacon_server.py: error: argument type: invalid choice: 'locale' (choose from 'core', 'local')
```
- <a href='https://github.com/syclops'></a> Ah, good point. Should I then use a simple if-else (and just check for `core`) or have an if-elif as above?
- <a href='https://github.com/kormat'></a> I think a simple if-else should be fine
- [x] <a href='#crh-comment-Pull 512f8ea9f15acbcc59ef180ff506be01b725943a infrastructure/beacon_server.py 4'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/275#discussion_r36078868'>File: infrastructure/beacon_server.py:L24-31</a></b>
- <a href='https://github.com/kormat'></a> Please keep imports sorted (with the 'import x' form coming before the 'from x import y' form). Same applies to other files.
- [x] <a href='#crh-comment-Pull 512f8ea9f15acbcc59ef180ff506be01b725943a infrastructure/dns_server.py 16'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/275#discussion_r36078943'>File: infrastructure/dns_server.py:L504-517</a></b>
- <a href='https://github.com/kormat'></a> Should be `'server_id'` for consistency
- [x] <a href='#crh-comment-Pull 512f8ea9f15acbcc59ef180ff506be01b725943a infrastructure/dns_server.py 17'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/275#discussion_r36078968'>File: infrastructure/dns_server.py:L504-517</a></b>
- <a href='https://github.com/kormat'></a> `'DNS Domain'` would be slightly clearer
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/275#issuecomment-127215473'>General Comment</a></b>
- <a href='https://github.com/kormat'></a> A few minor comments aside, very happy to see this change :)

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/275?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/275?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/275?approve=0'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/275'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: infrastructure/beacon_server.py [hunk 1]
```diff
@@ -1358,16 +1359,22 @@ def main():
...
+    if args.type == "core":
+        beacon_server = CoreBeaconServer(args.server_id, args.topo_file,
+                                         args.conf_file, args.path_policy_file)
+    elif args.type == "local":
+        beacon_server = LocalBeaconServer(args.server_id, args.topo_file,
+                                          args.conf_file,
+                                          args.path_policy_file)
     else:
         logging.error("First parameter can only be 'local' or 'core'!")
         sys.exit()
```
Comment (kormat) [inline infrastructure/beacon_server.py]: This isn't necessary any more:

```
beacon_server.py: error: argument type: invalid choice: 'locale' (choose from 'core', 'local')
```

File: infrastructure/beacon_server.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import sys
 import threading
 import time
+import argparse
```
Comment (kormat) [inline infrastructure/beacon_server.py]: Please keep imports sorted (with the 'import x' form coming before the 'from x import y' form). Same applies to other files.

File: infrastructure/beacon_server.py (hunk 1 above)
Comment (syclops) [inline infrastructure/beacon_server.py]: Ah, good point. Should I then use a simple if-else (and just check for `core`) or have an if-elif as above?

File: infrastructure/dns_server.py [hunk 3]
```diff
@@ -503,11 +504,13 @@ def main():
     """
     init_logging()
     handle_signals()
-    if len(sys.argv) != 4:
-        logging.error("run: %s server_id domain topo_file", sys.argv[0])
-        sys.exit()
+    parser = argparse.ArgumentParser()
+    parser.add_argument('server', help='Server identifier')
```
Comment (kormat) [inline infrastructure/dns_server.py]: Should be `'server_id'` for consistency

File: infrastructure/dns_server.py [hunk 4]
```diff
@@ -503,11 +504,13 @@ def main():
...
+    parser.add_argument('domain', help='Domain')
```
Comment (kormat) [inline infrastructure/dns_server.py]: `'DNS Domain'` would be slightly clearer

File: infrastructure/beacon_server.py (hunk 1 above)
Comment (kormat) [inline infrastructure/beacon_server.py]: I think a simple if-else should be fine

File: infrastructure/beacon_server.py (hunk 2 above)
Comment (kormat) [inline infrastructure/beacon_server.py]: :+1:<a href='#crh-update-none'></a>

File: infrastructure/dns_server.py (hunk 3 above)
Comment (kormat) [inline infrastructure/dns_server.py]: :+1:<a href='#crh-update-none'></a>

File: infrastructure/dns_server.py (hunk 4 above)
Comment (kormat) [inline infrastructure/dns_server.py]: :+1:<a href='#crh-update-none'></a>

File: infrastructure/beacon_server.py (hunk 1 above)
Comment (kormat) [inline infrastructure/beacon_server.py]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**